daemon/api_systems.go

```diff
@@ -99,6 +103,59 @@ func getSystems(c *Command, r *http.Request, user *auth.UserState) Response {
 	return SyncResponse(&rsp)
 }
 
+type oneSystemResponse struct {
+	// First part is designed to look like `client.System` - the
+	// only difference is how the model is represented
+	Current bool                   `json:"current,omitempty"`
+	Label   string                 `json:"label,omitempty"`
+	Model   map[string]interface{} `json:"model,omitempty"`
+	Actions []client.SystemAction  `json:"actions,omitempty"`
+	Brand   snap.StoreAccount      `json:"brand"`
+
+	// Volumes contains the volumes defined from the gadget snap
+	Volumes map[string]*gadget.Volume `json:"volumes,omitempty"`
+
+	// TODO: add "storage-encryption" via the
+	// devicestate.EncryptionSupportInfo() here too
+}
+
+// wrapped for unit tests
+var deviceManagerSystemAndGadgetInfo = func(dm *devicestate.DeviceManager, systemLabel string) (*devicestate.System, *gadget.Info, error) {
+	return dm.SystemAndGadgetInfo(systemLabel)
+}
+
+func getSystemDetails(c *Command, r *http.Request, user *auth.UserState) Response {
+	wantedSystemLabel := muxVars(r)["label"]
```

Comment:
Just asking because I don't know, if I call `/v2/systems//` can this end up being empty? And if so, should we check for an empty string?

Reply:
I think this is handled by the upper layers already, as a quick experiment I did try to access `/v2/systems//` and got nothing back at all. So it seems the underlying `gorilla/mux` is handling this already well.

Reply:
Cool, thank you for that!